src/components/project_card.rs

```diff
@@ -0,0 +1,155 @@
+/// - `insert_stars` - Will insert GitHub stars if the project has a `None` `star_count` property and a `Some(_)` `repository_url` property.
+/// If the repository URL is not a GitHub repository, this will fail.
+#[allow(non_snake_case)]
+#[inline_props]
+pub fn ProjectCard<'a>(cx: Scope, project: &'a Project<'a>, insert_stars: bool) -> Element {
+    render! {
+        div { class: "text-white p-4 bg-blue-1 rounded-md",
+            table { class: "text-left [&_th]:pr-4",
+                tr {
+                    th { "👀 Name" }
+                    td { "{project.name}" }
+                }
+                tr {
+                    th { "🛠️ Status" }
+                    td { "{project.status}" }
+                }
+                if let Some(description) = &project.description {
+                    rsx! {
+                        tr {
+                            th { "📜 Description" }
+                            td { "{description}" }
+                        }
+                    }
+                }
+                if let Some(repository_url) = &project.repository_url {
+                    rsx! {
+                        tr {
+                            th { "💾 Repository" }
+                            td { Link { class: "underline", to: "{repository_url}", "{repository_url}" } }
+                        }
+                    }
+                }
+                if let Some(website) = &project.website {
+                    if !website.is_empty() {
+                        rsx! {
+                            tr {
+                                th { "🌐 Website" }
+                                td { Link { class: "underline", to: "{website}", "{website}" } }
+                            }
+                        }
+                    } else {
+                        rsx! { "" }
+                    }
+                }
+                Stars { project: project, insert_stars: *insert_stars }
+            }
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+#[inline_props]
+fn Stars<'a>(cx: Scope, project: &'a Project<'a>, insert_stars: bool) -> Element {
+    if let Some(star_count) = project.star_count {
+        return render! {
+            tr {
+                th { "⭐ Stars" }
+                td { "{star_count}" }
+            }
+        };
+    }
+
+    if !insert_stars {
+        return None;
+    };
+
+    let Some(repository_url) = &project.repository_url else {
+        return None;
+    };
+
+    let fetched_stars = use_future(cx, &repository_url.to_string(), |url| {
+        fetch_star_count(url)
+    });
+
+    match fetched_stars.value() {
+        Some(Ok(star_count)) => render! {
+            tr {
+                th { "⭐ Stars" }
+                td { "{star_count}" }
+            }
+        },
+        Some(Err(e)) => {
+            log::error!("couldn't fetch stars from repository \"{repository_url}\". Error: {e}");
+
+            None
+        },
+        None => None,
```

Comment:
Could we maybe still show it as `⭐ Stars            ...` instead of being empty ? I think it will be better as it won't cause the layout to shift

Reply:
Yeah probably better that way.